device/thunder_ripple_sdk/src/client/thunder_client.rs

```diff
@@ -51,18 +53,108 @@ use ripple_sdk::{
     utils::error::RippleError,
 };
 use serde::{Deserialize, Serialize};
+use tokio::sync::oneshot::Sender;
 use url::Url;
 
-use crate::thunder_state::ThunderConnectionState;
-use crate::utils::get_error_value;
-
+use super::thunder_async_client::{ThunderAsyncClient, ThunderAsyncRequest, ThunderAsyncResponse};
 use super::thunder_client_pool::ThunderPoolCommand;
+use super::thunderbroker_plugins_status_mgr::{BrokerCallback, BrokerSender};
 use super::{
     jsonrpc_method_locator::JsonRpcMethodLocator,
     plugin_manager::{PluginActivatedResult, PluginManagerCommand},
 };
+use ripple_sdk::api::device::device_operator::{DeviceChannelRequest, DeviceResponseSubscription};
+use ripple_sdk::tokio::sync::mpsc::Receiver;
+use std::sync::RwLock;
 use std::{env, process::Command};
 
+pub type BrokerSubMap = HashMap<String, DeviceResponseSubscription>;
+pub type BrokerCallbackMap = HashMap<u64, Option<OneShotSender<DeviceResponseMessage>>>;
+
+#[derive(Debug)]
+pub struct ThunderClientManager;
+
+impl ThunderClientManager {
+    fn manage(
```

Comment:
Maybe rename this to `start()` to align with what you named the same for `ThunderAsyncClient`.